File: CSharp/Project/Talk/Quiz/FrameRate.cs
using System.Web;
using Talk.Required.Quiz;

namespace Talk.Quiz
{
    public class FrameRate
    {
        public void CreateCountdownFrames(HttpContext web, VideoMaker maker)
        {
            string type = web.Request.Params["TvFormat"];
            int frames = 0;

            if ("ntsc" == type)
            {
                frames = TvType.Ntsc.FrameRate.GetValueOrDefault();
            }
            for (int i = frames; i >= 0; i--)
            {
                maker.CreateFrame("" + i);
            }
        }
    }

    public class TvType
    {
        public static TvType Ntsc = new TvType();
        private static int? NtscFrameRame = 30;
        public int? FrameRate = NtscFrameRame;
    }
}

File: Java/java/src/com/spun/llewellyn/talks/legacycode/quiz/FrameRate.java
package com.spun.llewellyn.talks.legacycode.quiz;

import com.spun.llewellyn.talks.legacycode.required.VideoMaker;
import com.sun.net.httpserver.HttpContext;

public class FrameRate
{
  public void countdown(HttpContext web, VideoMaker maker)
  {
    String type = (String) web.getAttributes().get("TvFormat");
    Integer frame = 0;
    if ("ntsc" == type)
    {
      frame = TvType.NTSC.FrameRate;
    }
    do
    {
      maker.createFrame("" + frame);
      frame--;
    }
    while (frame > 0);
  }
}

class TvType
{
  public static TvType   NTSC            = new TvType();
  private static Integer NTSC_FRAME_RATE = 30;
  public Integer         FrameRate       = NTSC_FRAME_RATE;
}
